sabnzbd/downloader.py

```diff
@@ -771,45 +740,32 @@ def run(self):
                 continue
 
             if self.recv_threads > 1:
-                for nw, bytes_received, done in self.recv_pool.map(self.__recv, read):
+                futures = {self.recv_pool.submit(self.__process_nw, self.read_fds.get(fds)) for fds in read}
```

Comment:
This line is already a bit too cryptic for me ;)

Reply:
Hopefully clearer now 😄

I'm pretty close to being happy with the PR, I'll hopefull finish it tomorrow.